RFCs/0097-declarativeCDK

```diff
@@ -0,0 +1,48 @@
+# Declarative CDK
+
+* **Original Author(s):**: @evgenyka
+* **Tracking Issue**: #97
+
+# Summary
+
+Introducing Declarative CDK - a project that allows non-developer customers use CDK Constructs in their CloudFormation templates. 
+
+# Motivation
+
+The customers will benefit from an abstraction level that AWS CDK Constructs provide and ability to reuse existing Constructs content. It will result in the modularity of the CloudFormation templates, which will minimize the risks of ending up with template errors and also makes it easier to spot errors, if any.
+Customer can declare CDK Construct resource in the template, including all functions, dependencies, parameters and other settings. When CloudFormation is deploying the Stack from the template, the template will be pre-processed: the CDK App will be created and runtime by deCDK component that will be executed on the server side and the template will be synthesized on the fly. Then, the stack will be generated from that template by CloudFormation
+That process will allow to leverage CDK Constructs in CloudFormation template without any special knowledge of coding, exactly same way as declarative resources.
+
+# Example(s)
+
+See https://github.com/cdklabs/decdk/blob/main/examples/application-load-balancer.yaml
+
+# Limitation
+
+- Third-party libraries are not supported. Only aws-cdk-lib can be used.
+- User defined assets (e.g., code for a Lambda function) are not supported, as it runs on the server side. Assets that are bundled with the construct library, however, are supported. 
```

Comment:
We should call out that only local file assets are not supported. When file assets point to an S3 location, there is no problem in supporting them.

Reply:
Updated.